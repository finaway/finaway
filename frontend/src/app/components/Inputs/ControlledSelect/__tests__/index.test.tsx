import { render, renderHook } from '@testing-library/react';
import * as React from 'react';
import { useForm } from 'react-hook-form';
import { ControlledSelect } from '..';

interface FormValues {
  test: string;
}

const options = [
  { value: 1, label: 'Option 1' },
  { value: 2, label: 'Option 2' },
];

describe('<ControlledSelect  />', () => {
  const { result } = renderHook(() => useForm<FormValues>());
  const { control } = result.current;

  it('should match snapshot', () => {
    const loadingIndicator = render(
      <ControlledSelect
        control={control}
        name="test"
        label="Input Test"
        options={options}
      />,
    );
    expect(loadingIndicator.container.firstChild).toMatchSnapshot();
  });
});
