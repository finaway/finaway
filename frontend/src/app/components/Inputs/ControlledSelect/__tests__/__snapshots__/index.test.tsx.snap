// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ControlledSelect  /> should match snapshot 1`] = `
<div
  class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
>
  <label
    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink="false"
    for=""
  >
    Input Test
  </label>
  <div
    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="mui-component-select-test"
      class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      id="mui-component-select-test"
      role="button"
      tabindex="0"
    >
      <span
        class="notranslate"
      >
        â€‹
      </span>
    </div>
    <input
      aria-hidden="true"
      autocomplete="off"
      class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      data-testid=""
      name="test"
      tabindex="-1"
      value=""
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden="true"
      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        class="css-yjsfm1"
      >
        <span>
          Input Test
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
